jsondump/test.c

```diff
@@ -0,0 +1,70 @@
+#include <stdio.h>
+
+#include "jsondump.h"
+
+int main(int argc, char *argv[])
+{
+    struct json_dump *jd, *jd1;
+    
+    jd = jd_create(10);
+    jd_map_start(jd);
+    jd_map_add_string(jd, "foo", "bar");
+    jd_map_add_string(jd, "goo", "car");
+    jd_map_add_string(jd, "hoo", "\tdar\n");
+    jd_map_add_fmt(jd, "ioo", "%d ears", 2);
+    jd_map_add_null(jd, "nothing");
+    jd_map_add_bool(jd, "yes", 1);
+    jd_map_add_bool(jd, "no", 0);
+
+    printf("%s\n", jd->buf);
+
+    jd1 = jd_create(10);
+
+    jd_map_start(jd1);
+    jd_map_add_child(jd1, "nested", jd);
+    
+    printf("%s\n", jd1->buf);
+    
+    jd_destroy(jd);
+    jd_destroy(jd1);
+    
+    jd = jd_create(10);
```

Comment:
use a macro for 10 and in loop below

Reply:
It's not the same `10` though. The arg for `jd_create()` is the initial buffer size. The `10` in the `for` loop is just the number of elements we want in the list.